buildSrc/src/main/java/org/opensearch/gradle/testclusters/StandaloneRestIntegTestTask.java

```diff
@@ -133,6 +134,37 @@ public List<ResourceLock> getSharedResources() {
         return Collections.unmodifiableList(locks);
     }
 
+    private SharedResource getSharedResource(
+        BuildServiceRegistryInternal serviceRegistry,
+        Provider<TestClustersThrottle> throttleProvider
+    ) {
+        try {
+            try {
+                // The forService(Provider) is used by Gradle pre-8.0
+                return (SharedResource) MethodHandles.publicLookup()
+                    .findVirtual(
+                        BuildServiceRegistryInternal.class,
+                        "forService",
+                        MethodType.methodType(SharedResource.class, Provider.class)
+                    )
+                    .bindTo(serviceRegistry)
+                    .invokeExact(throttleProvider);
+            } catch (NoSuchMethodException | IllegalAccessException ex) {
+                // The forService(BuildServiceProvider) is used by Gradle post-8.0
+                return (SharedResource) MethodHandles.publicLookup()
+                    .findVirtual(
+                        BuildServiceRegistryInternal.class,
+                        "forService",
+                        MethodType.methodType(SharedResource.class, BuildServiceProvider.class)
+                    )
+                    .bindTo(serviceRegistry)
+                    .invokeExact((BuildServiceProvider<TestClustersThrottle, ?>) throttleProvider);
+            }
+        } catch (Throwable ex) {
+            throw new IllegalStateException("Unable to find suitable BuildServiceRegistryInternal::getResourceLock", ex);
```

Comment:
Copy/paste error on "getResourceLock" in exception message?

Reply:
Ahh.... thanks :)

Reply:
Fixed!